draco/analysis/flagging.py

```diff
@@ -1361,6 +1361,95 @@ def process(self, tstream, rfimask):
         return tsc
 
 
+class ApplyBaselineMask(task.SingleTask):
+    """Apply a distributed mask that varies across baselines.
+
+    No broadcasting is done, so the data and mask should have the same
+    axes. This shouldn't be used for non-distributed time-freq masks.
+
+    This task may produce output with shared datasets. Be warned that
+    this can produce unexpected outputs if not properly taken into
+    account.
+
+    Attributes
+    ----------
+    share : {"all", "none", "vis", "map"}
+        Which datasets should we share with the input. If "none" we create a
+        full copy of the data, if "vis" or "map" we create a copy only of the modified
+        weight dataset and the unmodified vis dataset is shared, if "all" we
+        modify in place and return the input container.
+    """
+
+    share = config.enum(["none", "vis", "map", "all"], default="all")
+
+    def process(self, data, mask):
+        """Flag data by zeroing the weights.
+
+        Parameters
+        ----------
+        data : timestream or sidereal stream
+            A timestream or sidereal stream like container. For example,
+            `containers.TimeStream`, `andata.CorrData` or
+            `containers.SiderealStream`.
+        mask : containers.BaselineMask  or containers.SiderealBaselineMask
+            An RFI mask for the same period of time.
+
+        Returns
+        -------
+        data : timestream or sidereal stream
+            The masked data. Masking is done in place.
+        """
+
+        if isinstance(mask, containers.BaselineMask):
+            if not hasattr(data, "time"):
+                raise TypeError(f"Expected a timestream-like type. Got {type(data)}.")
+
+            if not np.array_equal(data.time, mask.time):
+                raise ValueError("timestream and mask have different time axes.")
+
+        elif isinstance(mask, containers.SiderealBaselineMask):
+            if not hasattr(data, "ra"):
+                raise TypeError(
+                    f"Expected a sidereal stream like type. Got {type(data)}."
+                )
+
+            if not np.array_equal(data.ra, mask.ra):
+                raise ValueError("sidereal stream and mask have different RA axes.")
+
+        else:
+            raise TypeError(
+                f"Require a BaselineMask or SiderealBaselineMask. Git {type(mask)}."
+            )
+
+        # Validate remaining axes
+        if not np.array_equal(data.stack, mask.stack):
```

Comment:
`array_equal` is probably fine here, since the `stack` axis should be integers, but might as well change it for consistency:
```suggestion
        if not np.allclose(data.stack, mask.stack):
```